packages/api/src/command/webhook/webhook.ts

```diff
@@ -133,78 +147,112 @@ export const processRequest = async (
       sendAnalytics(status);
     }
     return true;
-
-    //eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    // TODO: 1411 - remove when DAPI is fully discontinued
+  } catch (error) {
+    // TODO: 1411 - remove the conditional and keep the code inside it when DAPI is fully discontinued
     if (productType === Product.medical) {
       log(`Failed to process WH request: ${errorToString(error)}`);
       const status = "failure";
-      try {
-        // mark this individual WH request as failed
-        await updateWebhookRequestStatus({
-          id: webhookRequest.id,
-          status,
-        });
-      } catch (err2) {
-        log(`Failed to store failure state on WH log: ${errorToString(err2)}`);
-        capture.error(err2, {
-          extra: {
-            webhookRequestId: webhookRequest.id,
-            webhookUrl,
-            context: `webhook.processRequest.updateStatus.failed`,
-            error: err2,
-          },
-        });
-      }
+      await Promise.all([
+        updateWhRequestWithError(error, webhookRequest.id, webhookUrl, status),
+        updateWhStatusWithError(error, webhookRequest.id, webhookUrl, settings.id),
+      ]);
       sendAnalytics(status);
-
-      let webhookStatusDetail;
-      if (error instanceof WebhookError) {
-        webhookStatusDetail = errorToWhStatusDetails(error);
-      } else {
-        log(`Unexpected error testing webhook`, error);
-        webhookStatusDetail = `Internal error: ${error.message}`;
-      }
-      try {
-        // update the status of the webhook endpoint on the cx's settings table
-        await updateWebhookStatus({
-          cxId: settings.id,
-          webhookEnabled: false,
-          webhookStatusDetail,
-        });
-      } catch (err2) {
-        log(`Failed to store failure state on WH settings: ${errorToString(err2)}`);
-        capture.error(err2, {
-          extra: {
-            webhookRequestId: webhookRequest.id,
-            webhookUrl,
-            context: `webhook.processRequest.updateStatus.details`,
-            error: err2,
-          },
-        });
-      }
```

Comment:
logic moved to `updateWhStatusWithError()`

Reply:
Diff:

<img width="1418" alt="image" src="https://github.com/metriport/metriport/assets/2132564/236606c2-7b04-4162-aa60-0910acfade7d">